### PR TITLE
[X86][DAG] Use nneg flag when trying to convert uitofp -> sitofp

### PR DESCRIPTION
**Pull request by goldsteinn**
- **[X86] Add tests for `uitofp nneg` -> `sitofp`; NFC**
- **[X86] Use `nneg` flag when trying to convert `uitofp` -> `sitofp`**

### CODE REVIEW

**Review by RKSimon — APPROVED** [review]
LGTM - cheers

File: llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp [hunk 1]
```diff
@@ -17382,8 +17382,9 @@ SDValue DAGCombiner::visitSINT_TO_FP(SDNode *N) {
   // but UINT_TO_FP is legal on this target, try to convert.
   if (!hasOperation(ISD::SINT_TO_FP, OpVT) &&
       hasOperation(ISD::UINT_TO_FP, OpVT)) {
+    SDNodeFlags Flags = N->getFlags();
     // If the sign bit is known to be zero, we can change this to UINT_TO_FP.
-    if (DAG.SignBitIsZero(N0))
+    if (Flags.hasNonNeg() || DAG.SignBitIsZero(N0))
```
Comment (RKSimon) [inline llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp:17387]: What targets can we test this with?

Comment (goldsteinn) [reply llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp:17387]: There are no failure in the test suite if I remove this code entirely...
I'm going to drop from this patch for now.

**Review by RKSimon — APPROVED** [review]
LGTM

**Review by dtcxzyw — APPROVED** [review]

**Pull request closed without merging**